### ISSUE TITLE
[Event Modification] codeunit 6620 "Copy Document Mgt."- OnAfterCopySalesHeaderDone

### ISSUE THREAD

**Issue by VMotin** [microsoft/ALAppExtensions#11717]
Hi, would it be possible to modify the event OnAfterCopySalesHeaderDone in codeunit 6620 to add an IsHandled, like this : 

```
            OnAfterCopySalesHeaderDone(
                ToSalesHeader, OldSalesHeader, FromSalesHeader, FromSalesShptHeader, FromSalesInvHeader,
                FromReturnRcptHeader, FromSalesCrMemoHeader, FromSalesHeaderArchive, IsHandled);

            if not IsHandled then begin
                Invoice := false;
                Ship := false;
            end;

```
```
    [IntegrationEvent(false, false)]
    local procedure OnAfterCopySalesHeaderDone(var ToSalesHeader: Record "Sales Header"; OldSalesHeader: Record "Sales Header"; FromSalesHeader: Record "Sales Header"; FromSalesShipmentHeader: Record "Sales Shipment Header"; FromSalesInvoiceHeader: Record "Sales Invoice Header"; FromReturnReceiptHeader: Record "Return Receipt Header"; FromSalesCrMemoHeader: Record "Sales Cr.Memo Header"; FromSalesHeaderArchive: Record "Sales Header Archive" ; var IsHandled : Boolean )
    begin
    end;

```
?
Thanks

**Comment by StanPesotskiy** [comment]
Why do you need Handled construction for updating fields Invoice and Ship of table ToSalesHeader if this variable is a parameter of event OnAfterCopySalesHeaderDone?

**Comment by VMotin** [comment]
After the event OnAfterCopySalesHeaderDone, the Invoice and Ship variables are set to False, so if I change them before, it will have no effect right ?

**Comment by StanPesotskiy** [comment]
In current code version we have this:
            OnAfterCopySalesHeaderDone( 
                ToSalesHeader, OldSalesHeader, FromSalesHeader, FromSalesShptHeader, FromSalesInvHeader,
                FromReturnRcptHeader, FromSalesCrMemoHeader, FromSalesHeaderArchive);

            ClearInvoiceAndShip(ToSalesHeader);

            if Status = Status::Released then begin
                Status := Status::Open;
                ReleaseDocument := true;
            end;

    local procedure ClearInvoiceAndShip(var ToSalesHeader: Record "Sales Header")
    var
        IsHandled: Boolean;
    begin
        IsHandled := false;
        OnBeforeClearInvoiceAndShip(ToSalesHeader, IsHandled);
        if IsHandled then
            exit;

        ToSalesHeader.Invoice := false;
        ToSalesHeader.Ship := false;
    end;

So, you can control Invoice and Ship by event OnBeforeClearInvoiceAndShip.

Seems this code has not been published yet and will be available for update 18.1

**Comment by AlexanderYakunin** [comment]
Closing
